### PR TITLE
Add application usage docs

### PR DESCRIPTION
**Pull request by endorama**
Provide more documentation around: how it works, install, verify, debug, upgrade

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+To install this application you should search for "Elastic Agent" in the GCP Marketplace and install it through the GCP Console.
+
+From the product listing you can choose the version of the application to install. We maintain `7.17.x` and `8.x` stack versions.
+
+By clicking "Configure" a guided process starts to collect some deploymenta details.
+
+You will need to choose:
+- the GKE cluster where to deploy the application: you can select an existing cluster or create a new one;
+- the namespace within the GKE cluster; selecting `default` will deploy the Application to the `default` namespace and the Elastic Agent in the (`kube-system`);
```
Comment (tommyers-elastic) [inline docs/installation.md:11]: minor formatting question - why the parentheses ? in `(`kube-system')`

File: docs/upgrading.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Upgrading your installed application
+
+Upgrading the application is **not supported**.
+
+Fleet UI does not allow to upgrade an Elastic Agent running under Kubernetes, as that would conflict with the orchestration performed by Kubernetes itself.
+
+GCP Marketplace does not support upgrading an application via its UI.
+
+## Workaround
```
Comment (tommyers-elastic) [inline docs/upgrading.md:9]: nit: i think we could improve the wording a little here. for example, upgrading the application isn't supported _in fleet_. and i don't think the other methods here should be considered workarounds, since they are the recommended way to upgrade. 

do you think out of the two options presented below we should recommend one over the other?

File: docs/upgrading.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+To workaround this there are 2 ways:
+1. once installed the Application behaves like any other Kubernetes resource. You can edit the application manifest (as you usually would with other Kubernetes resources). To do this, please refer to the Elastic Agent on Kubernetes upgrade documentation;
```
Comment (tommyers-elastic) [inline docs/upgrading.md:12]: can we link to the referenced docs?

**Review by tommyers-elastic — COMMENTED** [review]
awesome work

File: docs/deployer.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Deployer
 
-The deployer image is a Docker image that GCP will run in the customer Kubernetes cluster to the app's configuration to the Kubernetes API when users deploy the app from Cloud console.
+The `deployer` image is a Docker image that GCP will run in the customer GKE cluster to create all needed resources.
```
Comment (ravikesarwani) [inline docs/deployer.md:3]: Are we planning to bring this into the main “README.md” file? For example see: https://github.com/GoogleCloudPlatform/click-to-deploy/tree/master/k8s/wordpress. If that wasn’t the current plan I would highly suggest that we should bring all the different doc files into a single README.md itself rather than all these separate .md files.

Do we support Enthos as well? It would be good to clarify this in our document.

Can we add a simple diagram showing how the Elastic Agent is deployed and running in a very simple GKE cluster? It would be good for users to know if Elastic Agent runs as a DaemonSet etc. information?

File: docs/installation.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Installation
+
+To install this application you should search for "Elastic Agent" in the GCP Marketplace and install it through the GCP Console.
+
+From the product listing you can choose the version of the application to install. We maintain `7.17.x` and `8.x` stack versions.
+
```
Comment (ravikesarwani) [inline docs/installation.md:6]: Suggestion to add the following around what version to install => Review the [compatibility matrix](https://www.elastic.co/support/matrix#matrix_compatibility) for more details. Elastic recommends installing the latest version of the Elastic Agent that’s compatible with the version of your Elasticsearch deployment.

File: docs/installation.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+By clicking "Configure" a guided process starts to collect some deploymenta details.
```
Comment (ravikesarwani) [inline docs/installation.md:7]: Minor typo => deployment

File: docs/installation.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+You will need to choose:
+- the GKE cluster where to deploy the application: you can select an existing cluster or create a new one;
+- the namespace within the GKE cluster; selecting `default` will deploy the Application to the `default` namespace and the Elastic Agent in the (`kube-system`);
+- the application instance name (this is the name of the deployed application in your GKE cluster);
+- the Service Account of the application; **do not change this field**, a dedicated Service Account with required permissions will be created;
+- Fleet Server URL to connect to; if left empty Kibana Host, Kibana Fleet Username and Kibana Fleet Password are needed;
```
[the remark this answers is not in the document]
Comment (ravikesarwani) [reply docs/installation.md:14]: Can we provide some more details on where the users will find this? For example: Fleet Host URL can be found in Kibana->Management->Fleet->Settings.

File: docs/installation.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+- Fleet enrollment token; is an Elasticsearch API key to enroll one or more Elastic Agents in Fleet. See: https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html. If left empty Kibana Host, Kibana Fleet Username and Kibana Fleet Password are needed;
+- Kibana Username; user name to connect to Kibana;
+- Kibana Fleet Password; password to connect to Kibana;
+- Kibana Host; URL to connect to Kibana;
+- Container resource request - CPU; the requested container CPU, depends on use case;
```
[the remark this answers is not in the document]
Comment (ravikesarwani) [reply docs/installation.md:19]: Can we provide some more details on where the users will find these? For example: Fleet enrollment tokens can be found in Kibana->Management->Fleet->Enrollment tokens.
We have provided 2 methods. Is there a general recommendation we have for the users regarding using Fleet URL/token vs Kibana Host/Username/password

File: docs/installation.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+By clicking "Configure" a guided process starts to collect some deploymenta details.
+
+You will need to choose:
+- the GKE cluster where to deploy the application: you can select an existing cluster or create a new one;
+- the namespace within the GKE cluster; selecting `default` will deploy the Application to the `default` namespace and the Elastic Agent in the (`kube-system`);
+- the application instance name (this is the name of the deployed application in your GKE cluster);
+- the Service Account of the application; **do not change this field**, a dedicated Service Account with required permissions will be created;
+- Fleet Server URL to connect to; if left empty Kibana Host, Kibana Fleet Username and Kibana Fleet Password are needed;
+- Fleet enrollment token; is an Elasticsearch API key to enroll one or more Elastic Agents in Fleet. See: https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html. If left empty Kibana Host, Kibana Fleet Username and Kibana Fleet Password are needed;
+- Kibana Username; user name to connect to Kibana;
```
Comment (ravikesarwani) [inline docs/installation.md:16]: Is there specific permissions that the user needs to have? Can you please elaborate as appropriate.

File: docs/installation.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ravikesarwani) [reply docs/installation.md:19]: Let’s lean on the resource requirements with the work the Elastic Agent team is doing. Can we point users to https://www.elastic.co/guide/en/fleet/current/elastic-agent-installation.html for understanding the resource requirements?

File: docs/installation.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+- Kibana Fleet Password; password to connect to Kibana;
+- Kibana Host; URL to connect to Kibana;
```
Comment (ravikesarwani) [inline docs/installation.md:18]: Can we provide some more details on where the users will find Kibana URL? For example: Kibana URL can be found in the Elastic cloud console on the deployment details page.

File: docs/installation.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+- Container resource request - CPU; the requested container CPU, depends on use case;
+- Container Resource Request - Memory; the requested container memory, depends on use case.
+
+Once the application is installed it should automatically enroll in Fleet. 
+
+At this point the Elastic Agent is ready to accept policies, but is not yet actively collecting any data, as there is no default policy.
+
+To collect data go to the Fleet UI and configure an Agent Policy.
```
Comment (ravikesarwani) [inline docs/installation.md:26]: As I understand this may or may not be true? Say if the user already has an Agent policy and an associated Kubernetes integration and used its “token” wouldn’t the data start to get ingested automatically? I guess the real thing is what enrollment token has the user used? Feels like we need to provide some guidance to the users here? Can we expand on this in the document?

File: docs/next-steps.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
+# Next steps
+
+If you installed the application great, is now ready to collect data!
+
+By default the Elastic Agent does not have any policy assigned so will not collect any data.
+
+Here is a list of common use cases and how to setup Elastic Agent policies to collect releated data.
```
Comment (ravikesarwani) [inline docs/next-steps.md:7]: Minor typo => related

**Comment by tommyers-elastic** [comment]
FYI - docs ticket related to this here https://github.com/elastic/observability-docs/issues/2110

File: docs/upgrading.md (hunk 2 above)
Comment (endorama) [reply docs/upgrading.md:9]: > upgrading the application isn't supported _in fleet_.

Is actually not supported in GCP Marketplace too.

> i don't think the other methods here should be considered workarounds, since they are the recommended way to upgrade.

great observation, I will update the wording here to remove the current negative connotation.

> do you think out of the two options presented below we should recommend one over the other?

(1) is the usual K8s process but I really think is just a matter of preference (and availability of the enrollment token, needed for (2)). I will reword this to clarify.

File: docs/installation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md:11]: I think it's a leftover, good catch!

```suggestion
- the namespace within the GKE cluster; selecting `default` will deploy the Application to the `default` namespace and the Elastic Agent in the `kube-system` namespace;
```

File: docs/installation.md (hunk 11 above)
Comment (endorama) [reply docs/installation.md:26]: Correct, I didn't know about this! This is also mentioned in our [docs](https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html):
> All Elastic Agents enrolled through this token will use the selected policy unless you assign or enroll them in a different policy.

I'll add guidance on setting up the Kubernetes integration and use that enrollment token.

File: docs/next-steps.md (hunk 12 above)
Comment (endorama) [reply docs/next-steps.md:7]: Thanks for catching this

```suggestion
Here is a list of common use cases and how to setup Elastic Agent policies to collect related data.
```

File: docs/installation.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
+# Installation
```
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md:1]: > Is there an uninstall or delete step we need here as well (console & command line)?

A generic uninstall/delete or something specific to the installation step?

File: docs/installation.md (hunk 6 above)
Comment (endorama) [reply docs/installation.md:7]: Thanks for catching this!

```suggestion
By clicking "Configure" a guided process starts to collect some deployment details.
```

File: docs/installation.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md:14]: > would it be worth mentioning there that the alternative (kibana config) is for non-fleet managed agents?

@ravikesarwani do we want to support the standalone mode? Because if not we can remove Kibana host, username and password altogether.

File: docs/installation.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+At this point the Elastic Agent is ready to accept policies, but is not yet actively collecting any data, as there is no default policy.
+
+To collect data go to the Fleet UI and configure an Agent Policy.
+
+## Verify the setup
+
+Here are the steps to verify the installation worked as expected:
+1. ensure there are no errors in the installed Application page in the GCP Console;
+2. verify the Elastic Agent DaemonSet status, by clicking on it under Application Details; check Logs to ensure there are no errors;
+3. verify the Elastic Agent correctly enrolled with your Fleet instance;
```
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md]: I added a more context in https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/1/commits/67c1ed306f279deea2e4cb2f20f7e5d990eb3b37

Do you think is enough?

File: docs/installation.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+From the product listing you can choose the version of the application to install. We maintain `7.17.x` and `8.x` stack versions.
+
+By clicking "Configure" a guided process starts to collect some deploymenta details.
+
+You will need to choose:
+- the GKE cluster where to deploy the application: you can select an existing cluster or create a new one;
+- the namespace within the GKE cluster; selecting `default` will deploy the Application to the `default` namespace and the Elastic Agent in the (`kube-system`);
+- the application instance name (this is the name of the deployed application in your GKE cluster);
+- the Service Account of the application; **do not change this field**, a dedicated Service Account with required permissions will be created;
```
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md:16]: We need to as per [documentation](https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blob/master/docs/schema.md#type-service_account):
> All service accounts need to be defined as parameters in schema.yaml.
> If you add a Kubernetes ServiceAccount as a resource in your manifest, the deployment fails with an authentication error, because the deployer doesn't run with enough privileges to create a Service Account by default.

We can improve the wording though, and add a specific section about the service account and it's permissions in the documentation.

File: docs/deployer.md (hunk 4 above)
Comment (endorama) [reply docs/deployer.md:3]: > Are we planning to bring this into the main “README.md” file?

I wrote this documentation as contributors documentation, not end user documentation. A huge README makes contributing more difficult. I agree that for end users we should provide a more cohesive experience, I will discuss it with the docs team!

> Do we support Enthos as well? It would be good to clarify this in our document.

Yes, but I've not tested it. I will add it to the documentation.
(For future reference requirements are [here](https://cloud.google.com/marketplace/docs/partners/kubernetes/create-app-package#on-prem-reqs))

> Can we add a simple diagram showing how the Elastic Agent is deployed and running in a very simple GKE cluster? It would be good for users to know if Elastic Agent runs as a DaemonSet etc. information?

What about adding this information to the Elastic Agent on Kubernetes documentation? Given that the deployment structure is the same, I think would be better to have it in the general documentation than here as a special case.

File: docs/installation.md (hunk 5 above)
Comment (endorama) [reply docs/installation.md:9]: added!

File: docs/deployer.md (hunk 4 above)
Comment (endorama) [reply docs/deployer.md:3]: updated in https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/1/commits/de3efcdea01e14f062af2ac990c9bb4a184820d6

File: docs/upgrading.md (hunk 3 above)
Comment (endorama) [reply docs/upgrading.md]: updated in [`de3efcd` (#1)](https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/1/commits/de3efcdea01e14f062af2ac990c9bb4a184820d6)

File: docs/upgrading.md (hunk 2 above)
Comment (endorama) [reply docs/upgrading.md]: updated in [`de3efcd` (#1)](https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/1/commits/de3efcdea01e14f062af2ac990c9bb4a184820d6)

File: docs/installation.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (endorama) [reply docs/installation.md:17]: I opened an issue to discuss standalone mode support https://github.com/elastic/elastic-agent-gcp-kubernetes-app/issues/4

File: docs/installation.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply docs/installation.md]: looks good!

File: docs/upgrading.md (hunk 2 above)
Comment (tommyers-elastic) [reply docs/upgrading.md]: looks great@

**Review by tommyers-elastic — APPROVED** [review]

File: docs/installation.md (hunk 11 above)
Comment (endorama) [reply docs/installation.md:29]: I prepared https://github.com/elastic/elastic-agent-gcp-kubernetes-app/pull/7 to clarify this aspect.
